src/tools/rbd_mirror/Replayer.cc

```diff
@@ -93,21 +93,29 @@ int Replayer::init()
 void Replayer::run()
 {
   while (!m_stopping.read()) {
-    Mutex::Locker l(m_lock);
     set_sources(m_pool_watcher->get_images());
-    m_cond.WaitInterval(g_ceph_context, m_lock, seconds(30));
+    //m_cond.WaitInterval(g_ceph_context, m_lock, seconds(30));
```

Comment:
to summarize our irc discussion: this m_cond.WaitInterval() call is still needed here, so we don't call set_sources() continuously

Also, this commit is making set_sources() thread-safe, which is different from #14673, so the commit message should be updated to remove #14673